### PR TITLE
add new azure sql database alerts

### PR DESCRIPTION
**Pull request by yduartep**
add new azure sql database alerts

Screenshot 2024-11-14 at 14 59 27

### CODE REVIEW

File: csp-mixin/alerts/azure-alerts.yml [hunk 1]
```diff
@@ -22,9 +22,135 @@ groups:
...
+          service: 'Azure SQL database'
+          namespace: cloud-provider-azure
+        annotations:
+          summary: 'High database Storage usage.'
+          description: 'Archive or delete old data, or scale up storage capacity in {{ $labels.resourceName }} database.'
+          dashboard_uid: '82c5b6cf30db5b601c5cc3f5d8d4284d'
+
+      - alert: AzureDatabaseHighDeadlockCount
+        expr: |
+          sum by (job,resourceGroup,subscriptionName,resourceName) (azure_microsoft_sql_servers_databases_deadlock_total_count{job=~".+",resourceGroup=~".+",subscriptionName=~".+",resourceName=~".+"}) > 5
```
Comment (anaivanov) [inline csp-mixin/alerts/azure-alerts.yml:62]: I think we need a rate because this is a `count` metric

File: csp-mixin/alerts/azure-alerts.yml [hunk 2]
```diff
@@ -22,9 +22,135 @@ groups:
...
+          sum by (job,resourceGroup,subscriptionName,resourceName) (azure_microsoft_sql_servers_databases_connection_failed_total_count{job=~".+",resourceGroup=~".+",subscriptionName=~".+",resourceName=~".+"}) > 10
+        for: 5m
+        keep_firing_for: 10m
+        labels:
+          severity: warning
+          service: 'Azure SQL database'
+          namespace: cloud-provider-azure
+        annotations:
+          summary: 'High number of database System Failed connections.'
+          description: 'Check network problems, firewall restrictions or high resource consumption affecting application access to the database {{ $labels.resourceName }}.'
```
Comment (anaivanov) [inline csp-mixin/alerts/azure-alerts.yml:99]:
```suggestion
          description: 'Check network problems, firewall restrictions or high resource consumption affecting application access to the {{ $labels.resourceName }} database.'
```

File: csp-mixin/alerts/azure-alerts.yml (hunk 1 above)
Comment (yduartep) [reply csp-mixin/alerts/azure-alerts.yml:62]: yeah I think you are right. Thanks

File: csp-mixin/alerts/azure-alerts.yml [hunk 3]
```diff
@@ -22,9 +22,135 @@ groups:
...
+          service: 'Azure SQL database'
+          namespace: cloud-provider-azure
+        annotations:
+          summary: 'High database User CPU usage.'
+          description: 'Identify high CPU queries on {{ $labels.resourceName }} database and optimize them.'
+          dashboard_uid: '82c5b6cf30db5b601c5cc3f5d8d4284d'
+
+      - alert: AzureDatabaseHighSystemFailedConnections
+        expr: |
+          sum by (job,resourceGroup,subscriptionName,resourceName) (azure_microsoft_sql_servers_databases_connection_failed_total_count{job=~".+",resourceGroup=~".+",subscriptionName=~".+",resourceName=~".+"}) > 10
```
Comment (anaivanov) [inline csp-mixin/alerts/azure-alerts.yml:90]: Should this one also be a rate?

File: csp-mixin/alerts/azure-alerts.yml [hunk 4]
```diff
@@ -22,9 +22,135 @@ groups:
...
+          service: 'Azure SQL database'
+          namespace: cloud-provider-azure
+        annotations:
+          summary: 'High number of database System Failed connections.'
+          description: 'Check network problems, firewall restrictions or high resource consumption affecting application access to the {{ $labels.resourceName }} database.'
+          dashboard_uid: '82c5b6cf30db5b601c5cc3f5d8d4284d'
+
+      - alert: AzureDatabaseHighUserFailedConnections
+        expr: |
+          sum by (job,resourceGroup,subscriptionName,resourceName) (azure_microsoft_sql_servers_databases_connection_failed_user_error_total_count{job=~".+",resourceGroup=~".+",subscriptionName=~".+",resourceName=~".+"}) > 10
```
Comment (anaivanov) [inline csp-mixin/alerts/azure-alerts.yml:104]: and this one

**Review by anaivanov — APPROVED** [review]
LGTM 👏🏼

**Pull request merged**